ding/model/template/bcq.py

```diff
@@ -0,0 +1,108 @@
+from typing import Union, Dict, Optional, List
+from easydict import EasyDict
+import numpy as np
+import torch
+import torch.nn as nn
+
+from ding.utils import SequenceType, squeeze, MODEL_REGISTRY
+from ..common import RegressionHead, ReparameterizationHead
+from .vae import VanillaVAE
+
+
+@MODEL_REGISTRY.register('bcq')
+class BCQ(nn.Module):
+
+    mode = ['compute_actor', 'compute_critic', 'compute_vae', 'compute_eval']
+
+    def __init__(
+            self,
+            obs_shape: Union[int, SequenceType],
+            action_shape: Union[int, SequenceType, EasyDict],
+            actor_head_hidden_size: int = 64,
+            critic_head_hidden_size: int = 64,
+            activation: Optional[nn.Module] = nn.ReLU(),
+            norm_type: Optional[str] = None,
+            vae_hidden_dims: List = [750, 750],
+            phi: float = 0.05
+    ) -> None:
+        super(BCQ, self).__init__()
+        obs_shape: int = squeeze(obs_shape)
+        action_shape = squeeze(action_shape)
+        self.action_shape = action_shape
+        self.input_size = obs_shape
+        self.phi = phi
+
+        critic_input_size = self.input_size + action_shape
+        self.critic = nn.ModuleList()
+        for _ in range(2):
+            net = []
+            d = critic_input_size
+            for dim in critic_head_hidden_size:
+                net.append(nn.Linear(d, dim))
+                net.append(activation)
+                d = dim
+            net.append(nn.Linear(d, 1))
+            self.critic.append(nn.Sequential(*net))
+
+        net = []
+        d = critic_input_size
+        for dim in actor_head_hidden_size:
+            net.append(nn.Linear(d, dim))
+            net.append(activation)
+            d = dim
+        net.append(nn.Linear(d, 1))
+        self.actor = nn.Sequential(*net)
+
+        self.vae = VanillaVAE(action_shape, obs_shape, action_shape * 2, vae_hidden_dims)
+
+    def forward(self, inputs: Union[torch.Tensor, Dict[str, torch.Tensor]], mode: str) -> Dict[str, torch.Tensor]:
+        """
+        Overview:
+            The unique execution (forward) method of QAC method, and one can indicate different modes to implement \
+            different computation graph, including ``compute_actor`` and ``compute_critic`` in QAC.
+        Mode compute_actor:
+            Arguments:
+                - inputs (:obj:`torch.Tensor`): Observation data, defaults to tensor.
```

Comment:
arguments object type is Union[torch.Tensor, Dict[str, torch.Tensor]] for inputs?